File: Domain/Entities/Suggestion.cs

using System;
using System.Collections.Generic;

namespace SurveyApp.Domain.Entities
{
    public class Suggestion
    {
        public Guid Id { get; set; }
        public string Content { get; set; }
        public string CustomerName { get; set; }
        public string CustomerEmail { get; set; }
        public DateTime CreatedAt { get; set; }
        public SuggestionStatus Status { get; set; }
        public string Category { get; set; }
        public SuggestionPriority? Priority { get; set; }
        public bool IsAnonymous { get; set; }
        public string Response { get; set; }
        public DateTime? ResponseDate { get; set; }
        public List<string> SimilarSuggestions { get; set; } = new List<string>();
    }

    public enum SuggestionStatus
    {
        New,
        Reviewed,
        Implemented,
        Rejected
    }

    public enum SuggestionPriority
    {
        Low,
        Medium,
        High
    }
}

File: src/domain/entities/Suggestion.ts

export interface Suggestion {
  id: string;
  content: string;
  customerName: string;
  customerEmail: string;
  createdAt: string;
  status: 'new' | 'reviewed' | 'implemented' | 'rejected';
  category?: string;
  priority?: 'low' | 'medium' | 'high';
  isAnonymous?: boolean;
  response?: string;
  responseDate?: string;
  similarSuggestions?: string[];
}

export interface SuggestionFormData {
  content: string;
  customerName: string;
  customerEmail: string;
  category?: string;
  isAnonymous: boolean;
}

export interface MonthlyReport {
  month: string;
  year: number;
  totalSuggestions: number;
  implementedSuggestions: number;
  topCategories: {
    category: string;
    count: number;
  }[];
  suggestions: Suggestion[];
}
